src/core/surface/completion_queue.c

```diff
@@ -94,7 +94,7 @@ static void on_pollset_shutdown_done(grpc_exec_ctx *exec_ctx, void *cc,
 void grpc_cq_global_init(void) { gpr_mu_init(&g_freelist_mu); }
 
 void grpc_cq_global_shutdown(void) {
-  gpr_mu_destroy(&g_freelist_mu);
```

Comment:
Yes, that's correct. Your suggestion is better if there's a real problem. Mine was just papering over the problem because I believe it's a false positive. But this might lead to a real problem at some point in the future so the right answer is indeed to wait for all completion queues to be destroyed before doing the global shutdown.